lib/ecto/query/planner.ex

```diff
@@ -1799,6 +1799,10 @@ defmodule Ecto.Query.Planner do
 
   # Traverse all query components with expressions.
   # Therefore from, preload, assocs and lock are not traversed.
+  defp traverse_exprs(query, nil, acc, fun) do
```

Comment:
We should remove this clause. It should never be nil.

Reply:
Removed in https://github.com/elixir-ecto/ecto/pull/4220/commits/b1667db38f24b39090b36ae5373589de5b273896